### ISSUE TITLE
Remove pinned of chef 15 to latest chef-client

### ISSUE THREAD

**Issue by NAshwini** [chef-partners/azure-chef-extension#298]
### Issue Description
Currently azure-chef-extension by default pointing to `chef-15` which should be replaced by latest chef-client. But currently chef-16 is giving below error which need to fix.
Error:
```
2020-05-26 06:54:31 +0000 Creating chef configuration files
[2020-05-26T06:54:31+00:00] ERROR: undefined local variable or method `knife_config' for #<Chef::Knife::Core::BootstrapContext:0x00000000017cb6d0> (NameError)
(erubis):2:in `block in evaluate'
```

**Comment by NAshwini** [comment]
Currently tested azure-chef- extension with chef-client Version `16.3.38` but still facing the same.
Chef-client gets installed but facing error while creating chef configuration files for both Linux and windows. Need to handle it.

**Comment by trickyearlobe** [comment]
@NAshwini @Vasu1105 

This problem occurs because `Chef::Knife::Core::BootstrapContext.knife_config` was renamed to `Chef::Knife::Core::BootstrapContext.chef_config` in Chef Client 16.

It's used in
- https://github.com/chef-partners/azure-chef-extension/blob/master/lib/chef/azure/core/bootstrap_context.rb
- https://github.com/chef-partners/azure-chef-extension/blob/master/lib/chef/azure/core/windows_bootstrap_context.rb
- https://github.com/chef-partners/azure-chef-extension/blob/master/lib/chef/azure/bootstrap/chef-full.erb
- https://github.com/chef-partners/azure-chef-extension/blob/master/lib/chef/azure/bootstrap/windows-chef-client-msi.erb

It's likely fixed by PR #308 but that will need thoroughly testing as I have no access to publish Azure Extensions for testing myself.

**Comment by trickyearlobe** [comment]
It looks like the root cause of this problem is that Azure Chef Gem has its own custom bootstrap code (copied from Chef Client) that partially monkey patches `Chef::Knife::Core::BootstrapContext` that is shipped in Chef client (probably to override the bootstrap templates).

The version that ships in Chef client has diverged significantly due to some cleanup in several PR's such as https://github.com/chef/chef/pull/9646 but we haven't back-ported those changes into the Azure Chef Gem

If PR #308 gets us a quick fix for Chef Client 16, we should make a note to come back and fix this properly by getting rid of the monkey patching and supplying the templates in the proper manner.

**Comment by Vasu1105** [comment]
@NAshwini we don't have to work on 2nd point.

**Comment by NAshwini** [comment]
Delivered chef-16 with some minor fixes in azure chef extension version `1210.13.4.1`. Hence closing it.
